test/unit/EventAccumulator.spec.ts

```diff
@@ -0,0 +1,211 @@
+import { expect, sinon } from '../testUtils.js'
+import Countly from 'countly-sdk-nodejs'
+import { EventAccumulator } from '../../src/EventAccumulator.js'
+
+const sleep = async (ms: number): Promise<unknown> => await new Promise(resolve => setTimeout(resolve, ms))
+const sandbox = sinon.createSandbox()
+const addEventListenerStub = sandbox.stub()
+globalThis.addEventListener = addEventListenerStub
+
+describe('EventAccumulator', () => {
+  let countlyStub: sinon.SinonStubbedInstance<typeof Countly>
+
+  beforeEach(() => {
+    countlyStub = sandbox.stub(Countly)
```

Comment:
```suggestion
    countlyStub = sandbox.stub(Countly)
    clock = sinon.useFakeTimers()
```

we might also be able to call sandbox.useFakeTimers() and then benefit from `sandbox.restore()` restoring everything, but idk if that works